### ISSUE TITLE
S3 method `as.data.frame.proc_time` needs @export or   @exportS3method tag.

### ISSUE THREAD

**Issue by MatthieuStigler** [MatthieuStigler/matPkg#68]
From `document()`, get:

> 99_run_check.R:254: S3 method `as.data.frame.proc_time` needs @export or   @exportS3method tag.

But I am not sure I want to export it!?

Discussion:

- https://github.com/r-lib/roxygen2/issues/1592
- https://r-pkgs.org/dependencies-in-practice.html

**Comment by salim-b** [comment]
Terminology is a pain here.

`@export`ing a method like `as.data.frame.proc_time` does not actually export but only *register* it as an S3 method, i.e. add a line `S3method(as.data.frame,proc_time)` to the pkg `NAMESPACE`.

To actually export `as.data.frame.proc_time` (i.e. add a line `export(as.data.frame.proc_time)` to `NAMESPACE`), you'd need to add a separate `#' @export as.data.frame.proc_time` statement. Cf. [this comment](https://github.com/r-lib/roxygen2/issues/1322) by Gábor Csárdi.

tl;dr: Blindy trusting roxygen2's advice tends to be the right thing to do. 🤡

**Comment by MatthieuStigler** [comment]
hoi @salim-b 

thanks a lot for the explanations!
